File: app/src/main/java/com/example/myapplication/BankSampah.java
package com.example.myapplication;

import com.google.android.gms.maps.model.LatLng;

public class BankSampah {
    private String nama;
    private String alamat;
    private LatLng lokasi;
    private float jarak; // Menyimpan jarak dalam meter

    public BankSampah(String nama, String alamat, LatLng lokasi) {
        this.nama = nama;
        this.alamat = alamat;
        this.lokasi = lokasi;
        this.jarak = 0; // Default 0
    }

    public String getNama() {
        return nama;
    }

    public String getAlamat() {
        return alamat;
    }

    public LatLng getLokasi() {
        return lokasi;
    }

    public float getJarak() {
        return jarak;
    }

    public void setJarak(float jarak) {
        this.jarak = jarak;
    }
}

File: app/src/main/java/com/example/myapplication/BankSampah.kt
package com.example.myapplication

import com.google.android.gms.maps.model.LatLng
import com.google.firebase.database.Exclude

/**
 * Data class untuk merepresentasikan Bank Sampah.
 * Dibuat agar kompatibel dengan Firebase Realtime Database.
 */
data class BankSampah(
    // Properti yang langsung dipetakan ke field di Firebase
    var nama: String = "",
    var alamat: String = "",
    var latitude: Double = 0.0,
    var longitude: Double = 0.0,

    // Properti yang dikelola secara lokal di aplikasi dan tidak disimpan di Firebase
    @get:Exclude
    var id: String = "",
    @get:Exclude
    var jarak: Float = 0f
) {
    /**
     * Fungsi bantuan untuk mendapatkan objek LatLng dari data latitude dan longitude.
     * Anotasi @get:Exclude mencegah Firebase mencoba memetakan fungsi ini.
     */
    fun getLokasi(): LatLng {
        return LatLng(latitude, longitude)
    }
}
